src/subscription.ts

```diff
@@ -88,6 +88,20 @@ export type SetSubscriptionMetadataResponse = MetadataResponse;
 export type DetachSubscriptionCallback = EmptyCallback;
 export type DetachSubscriptionResponse = EmptyResponse;
 
+// JSDoc won't see these, so this is just to let you get typings
+// in your editor of choice. Callback types are open ended so that
+// not-quite-matching parameters won't cause a breaking change. We
+// may still shift these to required later.
+export declare interface Subscription {
+  on(event: 'message', listener: Function | ((message: Message) => void)): this;
+  on(event: 'error', listener: Function | ((error: Error) => void)): this;
+  on(event: 'close', listener: Function): this;
+
+  // Only used internally.
+  on(event: 'newListener', listener: Function): this;
```

Comment:
you could tighten this up more and declare the function signature:

`(event: string): void`

Reply:
I could combine a few of these, like:

```
on(event: 'close'|'newListener'|etc, listener: Function): this;
```

But in general they need to be broken out, since my intent is to show the available events and what their parameters are.

Reply:
I added a catch-all at the bottom that's impossible to match, to give somewhat better error messages. Otherwise the errors were all about `on('removeListener')` with it being at the bottom.